lib/main_drawer.dart

```diff
@@ -64,35 +62,23 @@ class MainDrawer extends StatelessWidget {
               iconData: Icons.home,
               navItemTitle: 'Home',
               callback: () {
-                Navigator.of(context).popUntil((route) => route.isFirst);
+                Navigator.of(context).pushReplacementNamed(
+                  RouteConstants.homeScreen,
+                );
               },
             ),
             DrawerNavItem(
-              iconData: Icons.stars_rounded,
-              navItemTitle: 'Starred Documents',
-              callback: () {
-                Navigator.of(context).pop();
-                Navigator.push(
-                    context,
-                    PageRouteBuilder(
-                      pageBuilder: (c, a1, a2) => Starred(),
-                      transitionsBuilder: (c, anim, a2, child) =>
-                          FadeTransition(opacity: anim, child: child),
-                      // transitionDuration: Duration(milliseconds: 2000),
-                    ));
-              },
-            ),
+                iconData: Icons.stars_rounded,
+                navItemTitle: 'Starred Documents',
+                callback: () {
+                  Navigator.of(context).pushReplacementNamed(
```

Comment:
Use push

Reply:
But why ? Before  this the app was using pop and push, which was ruining the whole transition, I just used the replacement which is doing the same thing, but giving us the proper effect.

Reply:
And If we use push on every drawer item, if user went on each screen through drawer, then it will create a large stack of screens which will effect the performance of the app and UX also (If user want to go back from the back button).

Reply:
Can you point out or show where the transitions were being ruined? The screen stack was cleared by pressing the home button in the drawer and the back button used to bring the user to the previous screen, @anushbhatia could you please confirm which approach should we use?

Reply:
@Saransh-cpp For example - When we click on starred documents button in the drawer, it pops the previous page and then pushes a new one. You can clearly compare and see the difference of transitions from both the approaches.